tests/test_thread.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+    tests/test_tread.cpp -- thread handling
+
+    Copyright (c) 2020 Nick Bridge <nick.bridge.chess@gmail.com>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#include "pybind11_tests.h"
+#include <thread>
+#include <atomic>
+#include <iostream>
+#include <pybind11/iostream.h>
+
+
+TEST_SUBMODULE(test_threading, m) {
+  // object to manage C++ thread
+  // simply repeatedly write to std::cerr until stopped
+  // redirect is called at some point to test the safety of scoped_estream_redirect
+  struct TestThread {
+    TestThread() : t_{nullptr}, stop_{false} {
+      auto thread_f = [this] {
```

Comment:
Oh, yes, and the convention is 4 spaces, I think. So that's why the formatting job on CI is complaining ;-)
And an empty line between methods etc might also help.

But I guess this will become obvious once you add it to a file with some existing code :-)

Reply:
Yeah - thanks. Was just fixing some of this. Hopefully I can get the hooks to work locally - I was having issues

Reply:
You should be able to install pre-commit, then run `pre-commit run -a`.

Reply:
I think pre-commit is running into firewall issues. I should be able to run most of it manually tho

Reply:
Or just match the rest of the formatting of `test_iostream.cpp`, and force-push any fixes that our CI still complains about :-)

Reply:
(Turnover time of this "Format" job should be reasonably minimal)